src/peep/peep.c

```diff
@@ -5661,7 +5661,7 @@ static void peep_update_thoughts(rct_peep* peep){
 					peep->window_invalidate_flags |= PEEP_INVALIDATE_PEEP_THOUGHTS;
 
 					// Clear top thought, push others up
-					if (i < PEEP_MAX_THOUGHTS - 1) {
+					if (i < PEEP_MAX_THOUGHTS - 2) {
```

Comment:
While in previous form it would still try to move 0 elements, it could cause a dereference of `peep->thoughts[5]` for `i == 4`